### ISSUE TITLE
upgrade-interactive freezes

### ISSUE THREAD

**Issue by jamiewinder** [yarnpkg/yarn#6198]
I've had this same problem on 1.9.2 and the previous stable version (1.7.0?).

I'm on Node 10.7.0 (also tried 10.6.0) on Windows 10.

If I run `yarn upgrade-interactive`, select the packages I want to upgrade, then I get the summary printed to the terminal, e.g. if I select react:
```bash
? Choose which packages to update. react@16.4.1
```
... but then it freezes. I can't abort (Ctrl-C) the command and return to the terminal either. 

I've tried running this with `--verbose` to give some more detail for this report, but unfortunately nothing else is printed to the terminal. Is there somewhere else I can look for an error to add to this report?

`upgrade`, `install` and all the other commands seem to work. I also doesn't just affect one project. I've also tried uninstalling and reinstalling yarn completely, to no avail.

**Comment by Jontem** [comment]
Same here on Windows 10, node v10.3.0 and yarn 1.7.0.

**Comment by jamiewinder** [comment]
Has anyone figured out a workaround to this while it gets sorted? I'm currently running the command to get the list of updatable dependencies then aborting out and doing the updates manually. Time consuming, but it works.

**Comment by chasecaleb** [comment]
I'm on Windows 10 with node 10.3.0 and yarn 1.9.4. I can confirm this behavior is consistently broken across cmd, powershell, and git-bash's mintty.

**Comment by Jordi-Carrasco** [comment]
Same here. Windows 10. node 10.7.0, yarn 1.9.4

**Comment by danielmoncada** [comment]
also having the same problem.

Windows 10, Yarn 1.9.4, Node 10.10.0

**Comment by j-cong** [comment]
I have this problem too.
Windows 8.1, node 10.6.0, yarn 1.10.1
With both git-bash and cmd

**Comment by beeplin** [comment]
yarn 1.10.1 solves this problem for me. windows 10 1709, node 8.12.0

**Comment by jamiewinder** [comment]
It's still the same for me on that version of Yarn unfortunately. 

Noticed a duplicate of this issue in #6367. 

It would be nice to be able to help with this, but I've no idea where to start. Are there any logs I can enable to help trace the cause? Like I said above, I've tried `--verbose` mode, but it doesn't give any extra info.

**Comment by worcode** [comment]
Same here

### env
- windows 10 build 1803
- node v10.12.0
- yarn 1.10.1

**Comment by bkniffler** [comment]
Hey there, I've just done `npm i -g yarn@1.12.0` and it seems to work.

**Comment by s-bauer** [comment]
@bkniffler Nope, does not seem to work for me! I tried yarn@1.12.0 and yarn@1.13.0 and both get stuck at the upgrade-interactive command!

**Comment by xaviergonz** [comment]
Same here

env
windows 10 build 1803
node v10.12.0
yarn 1.10.1 and yarn 1.12.0

**Comment by kziemski** [comment]
same here 


env
windows 10 build 1803
node v10.12.0
yarn 1.10.1 and yarn 1.12.0

**Comment by heyimalex** [comment]
I can reproduce this consistently. Console window just hangs indefinitely after you press enter after your choices are made.

- windows 10 build 16299.726
- node v10.11.0
- yarn HEAD

I couldn't figure it out. Stepping through in the chrome debugger, it freezes right after [this line](https://github.com/yarnpkg/yarn/blob/97834002c7fc1d487b68297ea05247a9ba8bbdf9/src/cli/commands/install.js#L271), which seemingly returns successfully but then the debugger just stops working. I'm pretty certain that line isn't the suspect, (and actually it runs fine with the exact same args earlier in the process), but that it suspending the event loop does it.

I ran with `--prof` and the bottom-up profile looks like this if it's helpful:

```
   ticks parent  name
  144940   98.9%  "C:/Program Files/nodejs/node.exe"
  144893  100.0%    "C:/Program Files/nodejs/node.exe"
  144408   99.7%      LazyCompile: ~handleWriteReq internal/stream_base_commons.js:12:24
  144408  100.0%        LazyCompile: ~writeGeneric internal/stream_base_commons.js:70:22
  144408  100.0%          LazyCompile: ~Socket._writeGeneric net.js:687:42
  144408  100.0%            LazyCompile: ~Socket._write net.js:724:35
```

I did find that replacing the `answers = await reporter.prompt` call [here](https://github.com/yarnpkg/yarn/blob/97834002c7fc1d487b68297ea05247a9ba8bbdf9/src/cli/commands/upgrade-interactive.js#L149) with a hard-coded value (same as the one returned from the call) made everything run just fine, so I *suspect* the issue lies there (I think in this case the console reporter). It looks like `upgrade-interactive` is the only place where `reporter.prompt` is ever called, and that select-options-from-the-list input is pretty fancy.
